goaviatrix/controller_bgp_max_as_limit_config.go

```diff
@@ -0,0 +1,74 @@
+package goaviatrix
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func (c *Client) SetControllerBgpMaxAsLimit(ctx context.Context, maxAsLimit int) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": fmt.Sprint(maxAsLimit),
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) DisableControllerBgpMaxAsLimit(ctx context.Context) error {
+	data := map[string]string{
+		"action":       "set_bgp_max_as_limit",
+		"CID":          c.CID,
+		"max_as_limit": "",
+	}
+
+	checkFunc := func(action, reason string, ret bool) error {
+		if !ret && !strings.Contains(reason, "Configured BGP maximum AS limit is not changed") {
+			return fmt.Errorf("rest API %s Post failed: %s", action, reason)
+		}
+		return nil
+	}
+	return c.PostAPIContext(ctx, data["action"], data, checkFunc)
+}
+
+func (c *Client) GetControllerBgpMaxAsLimit(ctx context.Context) (int, error) {
+	data := map[string]string{
+		"action": "show_bgp_max_as_limit",
+		"CID":    c.CID,
+	}
+
+	type BgpMaxAsLimitResponse struct {
+		Return bool   `json:"return"`
+		Reason string `json:"reason"`
+		Result string `json:"results"`
+	}
```

Comment:
You do not need the Return or Reason fields, GetAPIContext will already check those fields using your check function. 

Also if you name the `Result` field `Results` then you don't need the json struct tags either. By default, the `encoding/json` library will assume the json field name is the same as the struct field name in all lowercase.